server/build/index.js

```diff
@@ -2,12 +2,12 @@ import { join } from 'path'
 import fs from 'mz/fs'
 import uuid from 'uuid'
 import webpack from 'webpack'
-import getConfig from '../config'
+import getConfig, {phases} from '../config'
 import getBaseWebpackConfig from './webpack'
 import md5File from 'md5-file/promise'
 
 export default async function build (dir, conf = null) {
-  const config = getConfig(dir, conf)
+  const config = getConfig(phases.PRODUCTION_BUILD, dir, conf)
```

Comment:
How about using `next/constants` to get the `phase` as @rauchg suggested.

So, it could be like this:

```js
import { PHASE_PRODUCTION_BUILD } from 'next/constants'

module.exports = (phase, {defaultConfig}) => {
   if (phase === PHASE_PRODUCTION_BUILD) {
     // Do something
   }
   return {} // custom configuration
}
```